<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Intro to Metabarcoding using QIIME2 - Mammal Dataset: Importing, cleaning and quality control of the data</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/uom/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/uom/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/uom/favicon-16x16.png"><link rel="manifest" href="favicons/uom/site.webmanifest"><link rel="mask-icon" href="favicons/uom/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="Lesson Description" src="assets/images/uom-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/importing-cleaning-and-quality-control-of-the-data.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/uom-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Intro to Metabarcoding using QIIME2 - Mammal Dataset
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Intro to Metabarcoding using QIIME2 - Mammal Dataset
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Intro to Metabarcoding using QIIME2 - Mammal Dataset
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 6%" class="percentage">
    6%
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: 6%" aria-valuenow="6" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/importing-cleaning-and-quality-control-of-the-data.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="background.html">1. Background</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        2. Importing, cleaning and quality control of the data
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#import-data">Import Data</a></li>
<li><a href="#remove-primers">Remove Primers</a></li>
<li><a href="#create-and-interpret-sequence-quality-data">Create and interpret sequence quality data</a></li>
<li><a href="#denoising-the-data">Denoising the data</a></li>
<li><a href="#generate-summary-files">Generate Summary Files</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="taxonomic-analysis.html">3. Taxonomic Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="build-a-phylogenetic-tree.html">4. Build a phylogenetic tree</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="basic-visualisations-and-statistics.html">5. Basic visualisations and statistics</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="exporting-data-for-further-analysis-in-r.html">6. Exporting data for further analysis in R</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="extra-information.html">7. Extra Information</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="background.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="taxonomic-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="background.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Background
        </a>
        <a class="chapter-link float-end" href="taxonomic-analysis.html" rel="next">
          Next: Taxonomic Analysis...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Importing, cleaning and quality control of the data</h1>
        <p>Last updated on 2025-08-05 |

        <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/edit/main/episodes/importing-cleaning-and-quality-control-of-the-data.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do you import paired-end FASTQ files into QIIME2 using the
CasavaOneEight format?</li>
<li>Why is it important to remove primers before denoising, and how is
this done?</li>
<li>How can you assess sequence quality in QIIME2, and what features of
the quality plots help guide truncation parameters?</li>
<li>What is the purpose of the <code>dada2</code> denoise-paired
command, and what are some key parameters that influence its
output?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Import demultiplexed, paired ended sequencing data into QIIME2 using
the appropriate format and command structure</li>
<li>Explain why primer removal is necessary before denoising and apply
<code>cutadapt</code> command with appropriate primer sequences and
trimming parameters</li>
<li>Generate and interpret sequence quality viusalisations from
<code>qiime demux summarise</code> to determine suitable truncation
lengths, based on quality score drop offs</li>
<li>Summarise the function of <code>DADA</code>2 in QIIME2 and perform
denoising of paired-end sequences using the <code>DADA2</code> plugin,
adjusting key parameters like truncation length based on quality
data.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="import-data">Import Data<a class="anchor" aria-label="anchor" href="#import-data"></a></h2>
<hr class="half-width"><p><a href="https://www.melbournebioinformatics.org.au/tutorials/tutorials/qiime2_mammal/qiime2_mammal/#the-study" class="external-link">These
samples</a> were sequenced on a single Illumina MiSeq run using v3 (2 ×
300 bp) reagents at the Walter and Eliza Hall Institute (WEHI),
Melbourne, Australia. Data from WEHI came as paired-end, demultiplexed,
unzipped *.fastq files with adapters still attached. Following the <a href="https://docs.qiime2.org/2023.9/tutorials/importing/" class="external-link">QIIME2
importing tutorial</a>, this is the Casava One Eight format. The files
have been renamed to satisfy the Casava format as
SampleID_FWDXX-REVXX_L001_R[1 or 2]_001.fastq
e.g. CTRLA_Fwd04-Rev25_L001_R1_001.fastq.gz. The files were then zipped
(.gzip).</p>
<p>Here, the data files (two per sample i.e. forward and reverse reads
R1 and R2 respectively) will be imported and exported as a single QIIME
2 artefact file. These samples are already demultiplexed (i.e. sequences
from each sample have been written to separate files), so a metadata
file is not initially required.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>To check the input syntax for any QIIME2 command, enter the command,
followed by <code>--help</code>
e.g. <code>qiime tools import --help</code></p>
</div>
</div>
</div>
<p>Start by making a new directory analysis to store all the output
files from this tutorial. In addition, we will create a subdirectory
called seqs to store the exported sequences.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="bu">cd</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> analysis/seqs</span></code></pre>
</div>
<p>Run the command to import the raw data located in the directory
raw_data and export it to a single QIIME 2 artefact file,
combined.qza.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>--type <span class="st">'SampleData[PairedEndSequencesWithQuality]'</span> <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>--input-path raw_data <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>--input-format CasavaOneEightSingleLanePerSampleDirFmt <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>--output-path analysis/seqs/combined.qza</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="remove-primers">Remove Primers<a class="anchor" aria-label="anchor" href="#remove-primers"></a></h2>
<hr class="half-width"><p>These sequences still have the primers attached - they need to be
removed (using <code>cutadapt</code>) before denoising.</p>
<p>For this experiment, amplicons were amplified following the Earth
Microbiome protocol with 515F (Caporaso)– 806R (Caporaso) primers
targeting the v4 region of the 16S rRNA gene. The reads came back from
the sequencer with primers attached, which are removed before denoising
using <code>cutadapt</code> (v4.5 with python v3.8.15).</p>
<p>With <code>cutadapt</code>, the sequence specified and all bases
prior are trimmed; most sequences were trimmed at ~50 base pairs (bp).
An error rate of 0.15 was used to maximize the number of reads that the
primers were removed from while excluding nonspecific cutting. Any
untrimmed read was discarded.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">qiime</span> cutadapt trim-paired <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>--i-demultiplexed-sequences analysis/seqs/combined.qza <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>--p-front-f GTGCCAGCMGCCGCGGTAA <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>--p-front-r GGACTACHVGGGTWTCTAAT <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>--p-discard-untrimmed <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>--p-error-rate 0.15 <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>--output-dir analysis/seqs_trimmed <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<div id="tips-for-your-own-samples-and-analyses" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="tips-for-your-own-samples-and-analyses" class="callout-inner">
<h3 class="callout-title">Tips for Your Own Samples and Analyses</h3>
<div class="callout-content">
<ul><li>Remember to ask you sequencing facility if the raw data you get has
the primers attached - they may have already been removed.</li>
<li>The primers specified (515F (Caporaso)– 806R (Caporaso) targeting
the v4 region of the bacterial 16S rRNA gene) correspond to this
specific experiment - they will likely not work for your own data
analyses.</li>
<li>The error rate parameter, –p-error-rate, will likely need to be
adjusted for your own sample data to get 100% (or close to it) of reads
trimmed.</li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="create-and-interpret-sequence-quality-data">Create and interpret sequence quality data<a class="anchor" aria-label="anchor" href="#create-and-interpret-sequence-quality-data"></a></h2>
<hr class="half-width"><p>After trimming adapters and primers, it’s important to check the
overall quality of the sequence data before proceeding to denoising.
This step helps you make informed choices about where to trim or
truncate your reads and whether any samples should be excluded from
analysis.</p>
<p>To do this, we will generate a summary visualization of the
demultiplexed and trimmed data using QIIME2.</p>
<p>Create a subdirectory in <code>analysis</code> called
<code>visualisations</code> to store all files that we will visualise in
one place</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">mkdir</span> analysis/visualisations</span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">qiime</span> demux summarize <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>--i-data analysis/seqs_trimmed/trimmed_sequences.qza <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>--o-visualization analysis/visualisations/trimmed_sequences.qzv</span></code></pre>
</div>
<p>Copy the analysis/visualisations/trimmed_sequences.qzv file to your
local computer and <a href="https://view.qiime2.org/" class="external-link">view it using
QIIME 2 View</a>.</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Check Your Work: Read Quality and Demux Output </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fm484d0ukdn9n7a2gdw7hh%2Ftrimmed_sequences.qzv%3Frlkey%3Dta8bbyyuz6ucpyn4jymx78bp2%26dl%3D1" class="external-link">Click
to view the trimmed_sequence.qzv file in QIIME 2 View</a></p>
<ul><li>Make sure to switch between the “Overview” and “Interactive Quality
Plot” tabs in the top left hand corner.</li>
<li>Click and drag on the plot to zoom in. Double click to zoom back out
to full size.</li>
<li>Hover over a box to see the parametric seven-number summary of the
quality scores at the corresponding position.</li>
</ul></div>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Things To Look Out For In Quality Data </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<ul><li><p>Where does the median quality drop below 35? Look at the per-base
sequence quality plots for forward (R1) and reverse (R2) reads. Identify
the base pair position where the median quality score begins to sharply
decline. This informs your choices for –p-trunc-len-f and –p-trunc-len-r
in the next step.</p></li>
<li><p>Are there any samples with very low read counts? Under the
Interactive Sample Detail tab, check for samples with fewer than 1000
reads. These may not provide reliable data and might need to be excluded
later in R or at the diversity filtering stage.</p></li>
<li><p>Overall quality trends across reads It’s normal for reverse reads
(R2) to have lower quality toward the end. This may influence your
decision to truncate them earlier than forward reads.</p></li>
</ul></div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="denoising-the-data">Denoising the data<a class="anchor" aria-label="anchor" href="#denoising-the-data"></a></h2>
<hr class="half-width"><p>Trimmed sequences are now quality assessed using the
<code>dada2</code> <a href="https://pubmed.ncbi.nlm.nih.gov/27214047/" class="external-link">plugin within
QIIME2</a>. <code>dada2</code> denoises data by modelling and correcting
Illumina-sequenced amplicon errors, and infers exact amplicon sequence
variants (ASVs), resolving differences of as little as 1 nucleotide. Its
workflow consists of filtering, de-replication, reference‐free chimera
detection, and paired‐end reads merging, resulting in a feature or ASV
table.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>This step may long time to run (i.e. hours), depending on files sizes
and computational power.</p>
</div>
</div>
</div>
<div id="check-your-quality-assessment-understanding" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="check-your-quality-assessment-understanding" class="callout-inner">
<h3 class="callout-title">Check Your Quality Assessment Understanding</h3>
<div class="callout-content">
<p>Q: Based on your assessment of the quality plots from the
trimmed_sequences.qzv file generated in the previous step, what values
would you select for p-trunc-len-f and p-trunc-len-r in the command
below? Hint: Look for the position where the median quality score drops
below 35 for both forward and reverse reads?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>A: For version qiime2.2023.9 (other QIIME2 versions may slightly
differ), open trimmed_sequences.qzv file in QIIME2 view and select the
“Interactive Quality Plot” tab. Zoom in on the quality plots to
determine the base pair position where the median quality drops below
35: - Forward reads: ~253 bp - Reverse reads: ~208 bp</p>
<p>While these are the technical thresholds, it’s often better to
truncate more conservatively to improve quality and merging success. In
this workshop, we use: - p-trunc-len-f 213 - p-trunc-len-r 168</p>
<p>This trims an extra ~40 bp to ensure high-quality sequences are
retained.</p>
<p>Try experimenting with different truncation values and compare the
DADA2 denoising stats to see how your choice affects read retention and
quality.</p>
</div>
</div>
</div>
</div>
<p>In the following command, a pooling method of ‘pseudo’ is selected.
With the pseudo-pooling method, samples are denoised independently once,
ASVs detected in at least 2 samples are recorded, and samples are
denoised independently a second time, but this time with prior knowledge
of the recorded ASVs and thus higher sensitivity to those ASVs. This is
better than the default of ‘independent’ (where samples are denoised
independently) when you expect samples in the run to have similar ASVs
overall.</p>
<div id="note---workshop-participants-only" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="note---workshop-participants-only" class="callout-inner">
<h3 class="callout-title">Note - Workshop Participants Only</h3>
<div class="callout-content">
<p>Due to computational limitations in a workshop setting, this command
will be run staggered (by co-ordinating with other users on the Nectar
Instance you are logged in to), with no more than two users per Instance
running the command at the same time.</p>
</div>
</div>
</div>
<p>The specified output directory must not pre-exist.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">qiime</span> dada2 denoise-paired <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>--i-demultiplexed-seqs analysis/seqs_trimmed/trimmed_sequences.qza <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>--p-trunc-len-f xx <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>--p-trunc-len-r xx <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>--p-n-threads 0 <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>--p-pooling-method <span class="st">'pseudo'</span> <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>--output-dir analysis/dada2out <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>--verbose</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="generate-summary-files">Generate Summary Files<a class="anchor" aria-label="anchor" href="#generate-summary-files"></a></h2>
<hr class="half-width"><p>A <a href="https://docs.qiime2.org/2023.9/tutorials/metadata/" class="external-link">metadata
file</a> is required which provides the key to gaining biological
insight from your data. The file echidna_metadata.tsv is provided in the
home directory of your Nectar instance. This spreadsheet has already
been verified using the plugin for Google Sheets, <a href="https://keemei.qiime2.org/" class="external-link">keemei</a>.</p>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Things To Look Out For In Summary Data </h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" aria-labelledby="headingSpoiler3" data-bs-parent="#accordionSpoiler3">
<div class="accordion-body">
<ul><li><p>How many features (ASVs) were generated? Are the communities high
or low diversity?</p></li>
<li><p>Do BLAST searches of the representative sequences make sense? Are
the features what you would expect e.g. marine or terrestrial?</p></li>
<li><p>Have a large number (e.g. &gt;50%) of sequences been lost during
denoising/filtering? If so, the settings might be too
stringent.</p></li>
</ul></div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">qiime</span> metadata tabulate <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>--m-input-file analysis/dada2out/denoising_stats.qza <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>--o-visualization analysis/visualisations/16s_denoising_stats.qzv <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<p>Copy analysis/visualisations/16s_denoising_stats.qzv to your local
computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler4" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler4" aria-expanded="false" aria-controls="collapseSpoiler4">
  <h3 class="accordion-header" id="headingSpoiler4">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Check Your Work: Denoising Stats and Visualisation </h3>
</button>
<div id="collapseSpoiler4" class="accordion-collapse collapse" aria-labelledby="headingSpoiler4" data-bs-parent="#accordionSpoiler4">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fhwpyu01wm0ubb82wbb1dt%2F16s_denoising_stats.qzv%3Frlkey%3Dkuc0zfuozlpzr0c49sxhr96m7%26dl%3D1" class="external-link">Click
to view the 16s_denoising_stats.qzv file in QIIME 2 View</a></p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table summarize <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>--i-table analysis/dada2out/table.qza <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>--m-sample-metadata-file echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>--o-visualization analysis/visualisations/16s_table.qzv <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<p>Copy analysis/visualisations/16s_table.qzv to your local computer and
view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler5" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler5" aria-expanded="false" aria-controls="collapseSpoiler5">
  <h3 class="accordion-header" id="headingSpoiler5">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Check Your Work: Feature/ASV Summary and Visualisation </h3>
</button>
<div id="collapseSpoiler5" class="accordion-collapse collapse" aria-labelledby="headingSpoiler5" data-bs-parent="#accordionSpoiler5">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fhkvk2yqpnm4w9oy64maf3%2F16s_table.qzv%3Frlkey%3Dtm368u59rexzgdwpm9m6l11l1%26dl%3D1" class="external-link">Click
to view the 16s_table.qzv file in QIIME 2 View</a></p>
<p>Make sure to switch between the “Overview” and “Feature Detail” tabs
in the top left hand corner.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table tabulate-seqs <span class="dt">\</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>--i-data analysis/dada2out/representative_sequences.qza <span class="dt">\</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>--o-visualization analysis/visualisations/16s_representative_seqs.qzv <span class="dt">\</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<p>Copy analysis/visualisations/16s_representative_seqs.qzv to your
local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler6" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler6" aria-expanded="false" aria-controls="collapseSpoiler6">
  <h3 class="accordion-header" id="headingSpoiler6">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Check Your Work: Representative Sequences and Visualisation </h3>
</button>
<div id="collapseSpoiler6" class="accordion-collapse collapse" aria-labelledby="headingSpoiler6" data-bs-parent="#accordionSpoiler6">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2F2uesky0zntkapxhlr58gu%2F16s_representative_seqs.qzv%3Frlkey%3Dakmmi05wpcwommjuytgub2grk%26dl%3D1" class="external-link">Click
here to view the 16s_representative_seqs.qzv file in QIIME 2
View</a></p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>QIIME2 can import demultiplexed paired-end reads as a single
artefact using the CasavaOneEight format.</li>
<li>Primer removal with <code>cutadapt</code> and inspection of quality
plots helps determine optimal trimming parameters before denoising.</li>
<li>DADA2 denoises and merges paired reads to generate high-quality
ASVs, which can be summarised and evaluated using QIIME2
visualisations.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="background.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="taxonomic-analysis.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="background.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Background
        </a>
        <a class="chapter-link float-end" href="taxonomic-analysis.html" rel="next">
          Next: Taxonomic Analysis...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/edit/main/episodes/importing-cleaning-and-quality-control-of-the-data.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/" class="external-link">Source</a></p>
				<p><a href="https://github.com/rpapaluca/intro-QIIME2-mammal/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:fixme@fixme.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.12" class="external-link">sandpaper (0.16.12)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7.9000)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/importing-cleaning-and-quality-control-of-the-data.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "metabarcoding, QIIME2, microbiome ",
  "name": "Importing, cleaning and quality control of the data",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/importing-cleaning-and-quality-control-of-the-data.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/importing-cleaning-and-quality-control-of-the-data.html",
  "dateCreated": "2023-11-01",
  "dateModified": "2025-08-05",
  "datePublished": "2025-08-05"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

