<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Intro to Metabarcoding using QIIME2 - Mammal Dataset: Section 4: Basic visualisations and statistics</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/uom/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/uom/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/uom/favicon-16x16.png"><link rel="manifest" href="../favicons/uom/site.webmanifest"><link rel="mask-icon" href="../favicons/uom/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav uom"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="uom-logo" alt="Lesson Description" src="../assets/images/uom-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../section-4-basic-visualisations-and-statistics.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav uom" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/uom-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Intro to Metabarcoding using QIIME2 - Mammal Dataset
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Intro to Metabarcoding using QIIME2 - Mammal Dataset
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Intro to Metabarcoding using QIIME2 - Mammal Dataset
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 80%" class="percentage">
    80%
  </div>
  <div class="progress uom">
    <div class="progress-bar uom" role="progressbar" style="width: 80%" aria-valuenow="80" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../section-4-basic-visualisations-and-statistics.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Using Markdown</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="background.html">2. Background</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="section-1-importing-cleaning-and-quality-control-of-the-data.html">3. Section 1: Importing, cleaning and quality control of the data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="section-2-taxonomic-analysis.html">4. Section 2: Taxonomic Analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="section-3-build-a-phylogenetic-tree.html">5. Section 3: Build a phylogenetic tree</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        6. Section 4: Basic visualisations and statistics
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#visualising-asv-relative-abundance">Visualising ASV Relative Abundance</a></li>
<li><a href="#exploring-sequencing-depth-rarefaction-curves">Exploring Sequencing Depth (Rarefaction Curves)</a></li>
<li><a href="#core-diversity-metrics-alpha-and-beta-diversity">Core Diversity Metrics (Alpha and Beta Diversity)</a></li>
<li><a href="#testing-for-group-differences">Testing for Group Differences</a></li>
<li><a href="#provenance-and-reproducibility">Provenance and Reproducibility</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="section-5-exporting-data-for-further-analysis-in-r.html">7. Section 5 - Exporting data for further analysis in R</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="section-6-extra-information.html">8. Section 6 - Extra Information</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/section-3-build-a-phylogenetic-tree.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/section-5-exporting-data-for-further-analysis-in-r.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/section-3-build-a-phylogenetic-tree.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Section 3: Build a
        </a>
        <a class="chapter-link float-end" href="../instructor/section-5-exporting-data-for-further-analysis-in-r.html" rel="next">
          Next: Section 5 -...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Section 4: Basic visualisations and statistics</h1>
        <p>Last updated on 2025-08-05 |

        <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/edit/main/episodes/section-4-basic-visualisations-and-statistics.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can you use QIIME2 to visualise the relative abundance of taxa
across samples?</li>
<li>What do rarefaction curves tell you about sequencing depth and
sample diversity?</li>
<li>What’s the difference between alpha and beta diversity, and how are
these metrics calculated and interpreted in QIIME2?</li>
<li>How do you test for statistical differences in diversity or
community composition between sample groups using QIIME2
visualisations?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Generate relative abundance bar plots using the qiime taxa barplot
command and explore patterns across metadata categories by changing
taxonomic levels and sorting options in QIIME 2 View.</li>
<li>Interpret alpha rarefaction curves to evaluate whether sequencing
depth is sufficient to capture microbial community diversity across
samples.</li>
<li>Explain the purpose of core diversity metrics (e.g., Shannon
diversity, UniFrac distances), and apply the core-metrics-phylogenetic
method to generate interactive visualisations for comparing diversity
across samples.</li>
<li>Assess group-level differences in alpha and beta diversity using
appropriate statistical tests (e.g., PERMANOVA), and interpret QIIME2
outputs to draw biological insights from metadata categories.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="visualising-asv-relative-abundance">Visualising ASV Relative Abundance<a class="anchor" aria-label="anchor" href="#visualising-asv-relative-abundance"></a></h2>
<hr class="half-width"><p>We can create bar charts to compare the relative abundance of ASVs
across samples.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>--i-table analysis/taxonomy/16s_table_filtered.qza <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>--i-taxonomy analysis/taxonomy/classification.qza <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>--m-metadata-file echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>--o-visualization analysis/visualisations/barchart.qzv <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<p>Copy <code>analysis/visualisations/barchart.qzv</code> to your local
computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Taxonomy Barplots
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2F0r73637tkg60wjbf5s07a%2Fbarchart.qzv%3Frlkey%3Djlx2ctcs9n5rszkun7bgaint1%26dl%3D1" class="external-link">Click
to view the barchart.qzv file in QIIME 2 View</a></p>
</div>
</div>
</div>
</div>
<div id="how-to-explore-barplots-in-qiime-2-view" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-explore-barplots-in-qiime-2-view" class="callout-inner">
<h3 class="callout-title">How to Explore Barplots in QIIME 2 View</h3>
<div class="callout-content">
<p>Once you’ve opened barchart.qzv in QIIME 2 View, try the
following:</p>
<ul><li>
<em>Change Taxonomic Levels</em>: Use the “Change Taxonomic Level”
dropdown to explore patterns at Level 1 (Phylum), Level 3 (Class), and
Level 5 (Family).</li>
<li>
<em>Sort Samples</em>: Use the “Sort Samples By” option to group by
metadata fields like Sex or Animal.</li>
<li>
<em>Adjust Visuals</em>: Increase the “Bar Width” to make
differences easier to see across samples.</li>
</ul><p>These visualisations can help you spot differences in microbiome
composition across individuals or groups.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="exploring-sequencing-depth-rarefaction-curves">Exploring Sequencing Depth (Rarefaction Curves)<a class="anchor" aria-label="anchor" href="#exploring-sequencing-depth-rarefaction-curves"></a></h2>
<hr class="half-width"><p>Rarefaction curves help determine whether the samples have been
sequenced deeply enough to capture all the community members.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>How to Explore Rarefaction Curves in QIIME2 View</strong>: -
Do the curves plateau? If not, you may not have sequenced deeply enough.
- What sequencing depth corresponds to most curves levelling off? -
Choose a <code>--p-max-depth value</code> based on
<code>16s_table_filtered.qzv</code>. A value near the median frequency
works well.</p>
</div>
</div>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Further Detail on -p-max-depth
  </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<p>The value that you provide for –p-max-depth should be determined by
reviewing the “Frequency per sample” information presented in the
<code>16s_table_filtered.qzv</code> file that was created above. In
general, choosing a value that is somewhere around the median frequency
seems to work well, but you may want to increase that value if the lines
in the resulting rarefaction plot don’t appear to be levelling out, or
decrease that value if you seem to be losing many of your samples due to
low total frequencies closer to the minimum sampling depth than the
maximum sampling depth.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-rarefaction <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>--i-table analysis/taxonomy/16s_table_filtered.qza <span class="dt">\</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>--i-phylogeny analysis/tree/16s_rooted_tree.qza <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>--p-max-depth 20000 <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>--p-min-depth 500 <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>--p-steps 40 <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>--m-metadata-file echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>--o-visualization analysis/visualisations/16s_alpha_rarefaction.qzv <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>--verbose</span></code></pre>
</div>
<p>Copy <code>analysis/visualisations/16s_alpha_rarefaction.qzv</code>
to your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler3" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler3" aria-expanded="false" aria-controls="collapseSpoiler3">
  <h3 class="accordion-header" id="headingSpoiler3">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Rarefaction Visualisation
  </h3>
</button>
<div id="collapseSpoiler3" class="accordion-collapse collapse" aria-labelledby="headingSpoiler3" data-bs-parent="#accordionSpoiler3">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fa9tqrizdol82j4f2em7ll%2F16s_alpha_rarefaction.qzv%3Frlkey%3Dilcp14psote78zp7rbjrdlz26%26dl%3D1" class="external-link">Click
to view the 16s_alpha_rarefaction.qzv file in QIIME 2 View</a></p>
<p>Select “Animal” in the “Sample Metadata Column” and
“observed_features” under “Metric”</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="core-diversity-metrics-alpha-and-beta-diversity">Core Diversity Metrics (Alpha and Beta Diversity)<a class="anchor" aria-label="anchor" href="#core-diversity-metrics-alpha-and-beta-diversity"></a></h2>
<hr class="half-width"><div class="section level3">
<h3 id="what-is-core-diversity-analysis">What is Core Diversity Analysis?<a class="anchor" aria-label="anchor" href="#what-is-core-diversity-analysis"></a></h3>
<p>QIIME 2’s <code>q2-diversity</code> plugin allows us to explore alpha
(within-sample) and beta (between-sample) diversity using a range of
metrics and visualisation tools. The following is adapted from the
Moving Pictures Tutorial.</p>
<p>The <code>core-metrics-phylogenetic</code> pipeline simplifies this
by:</p>
<ul><li>Rarefying the feature table to a fixed sampling depth.</li>
<li>Calculating several diversity metrics.</li>
<li>Generating PCoA plots for beta diversity using Emperor.</li>
</ul><p><strong>Alpha diversity metrics (within samples):</strong></p>
<ul><li>
<em>Observed OTUs</em>: a qualitative measure of community
richness</li>
<li>
<em>Shannon’s index</em>: a quantitative measure of community
richness</li>
<li>
<em>Faith’s Phylogenetic Diversity</em>: a qualitative measure of
community richness that considers phylogenetic relatedness</li>
<li>
<em>Evenness (Pielou’s)</em>: measure of community evenness</li>
</ul><p><strong>Beta diversity metrics (between samples):</strong></p>
<ul><li>
<em>Jaccard distance</em>: Qualitative measure of community
dissimilarity, based on absence/presence</li>
<li>
<em>Bray–Curtis distance</em>: Quantitative measure of community
dissimilarity, based on abundance</li>
<li>
<em>Unweighted UniFrac</em>: Qualitative measure of community
dissimilarity, includes phylogenetic relationships between features</li>
<li>
<em>Weighted UniFrac</em>: Quantitative measure of community
dissimilarity, includes phylogenetic relationships between features</li>
</ul><p><strong>Choosing your sampling depth</strong></p>
<p>Most diversity metrics are sensitive to sampling depth, so QIIME 2
standardizes counts by randomly subsampling the counts from each sample
to the value provided for this parameter. This is passed using
<code>--p-sampling-depth</code>.</p>
<p>For example, if <code>--p-sampling-depth 500</code> is provided, this
step will subsample the counts in each sample without replacement, so
that each sample in the resulting table has a total count of 500. If the
total count for any sample(s) are smaller than this value, those samples
will be excluded from the diversity analysis.</p>
<p>Choosing this value is tricky. To do so, keep in mind the
following:</p>
<ul><li>Review the “Frequency per sample” in
<code>16s_table_filtered.qzv</code> created previously</li>
<li>Pick a depth that retains most samples while still being high enough
to capture diversity.</li>
<li>It’s okay to omit extraction blanks or PCR negatives.</li>
</ul></div>
<div class="section level3">
<h3 id="calculating-core-diversity-metrics">Calculating Core Diversity Metrics<a class="anchor" aria-label="anchor" href="#calculating-core-diversity-metrics"></a></h3>
<p>Once you’ve determined an appropriate depth, you can run the
following command to generate a full suite of diversity metrics:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">qiime</span> diversity core-metrics-phylogenetic <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="at">--i-phylogeny</span> analysis/tree/16s_rooted_tree.qza <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">--i-table</span> analysis/taxonomy/16s_table_filtered.qza <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">--p-sampling-depth</span> 7316 <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">--output-dir</span> analysis/diversity_metrics</span></code></pre>
</div>
<p>Copy the <code>.qzv</code> files created from the above command into
the <code>visualisations</code> subdirectory.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">cp</span> analysis/diversity_metrics/<span class="pp">*</span>.qzv analysis/visualisations</span></code></pre>
</div>
<p>To view the differences between sample composition using unweighted
UniFrac in ordination space, copy
<code>analysis/visualisations/unweighted_unifrac_emperor.qzv</code> to
your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler4" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler4" aria-expanded="false" aria-controls="collapseSpoiler4">
  <h3 class="accordion-header" id="headingSpoiler4">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Unweighted UniFrac Emperor Ordination
  </h3>
</button>
<div id="collapseSpoiler4" class="accordion-collapse collapse" aria-labelledby="headingSpoiler4" data-bs-parent="#accordionSpoiler4">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fpsqxp4hfgvrzpeta687lk%2Funweighted_unifrac_emperor.qzv%3Frlkey%3Dujflsjcjx5qjl2z9vc61072pi%26dl%3D1" class="external-link">Click
to view the unweighted_unifrac_emperor.qzv file in QIIME 2 View</a></p>
<p>On q2view, select the “Color” tab, choose “Animal” under the “Select
a Color Category” dropdown menu, then select the “Shape” tab and choose
“Sex” under the “Select a Shape Category” dropdown menu.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="testing-for-group-differences">Testing for Group Differences<a class="anchor" aria-label="anchor" href="#testing-for-group-differences"></a></h2>
<hr class="half-width"><p>Next, we’ll test for association between categorical metadata columns
and alpha diversity data. We’ll do that here for observed ASVs and
evenness metrics.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">--i-alpha-diversity</span> analysis/diversity_metrics/observed_features_vector.qza <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">--o-visualization</span> analysis/visualisations/observed_features-significance.qzv</span></code></pre>
</div>
<p>Copy
<code>analysis/visualisations/observed_features-significance.qzv</code>
to your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler5" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler5" aria-expanded="false" aria-controls="collapseSpoiler5">
  <h3 class="accordion-header" id="headingSpoiler5">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Observed Diversity Output
  </h3>
</button>
<div id="collapseSpoiler5" class="accordion-collapse collapse" aria-labelledby="headingSpoiler5" data-bs-parent="#accordionSpoiler5">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Ftn3r31tk3s04pv19cb0ig%2Fobserved_features-significance.qzv%3Frlkey%3D0qmb7fezh6am8drib39qn629h%26dl%3D1" class="external-link">Click
to view the observed_features-significance.qzv file in QIIME 2
View.</a></p>
<p>Select “Animal” under the “Column” dropdown menu.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">--i-alpha-diversity</span> analysis/diversity_metrics/evenness_vector.qza <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">--o-visualization</span> analysis/visualisations/evenness-group-significance.qzv</span></code></pre>
</div>
<p>Copy
<code>analysis/visualisations/evenness-group-significance.qzv</code> to
your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler6" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler6" aria-expanded="false" aria-controls="collapseSpoiler6">
  <h3 class="accordion-header" id="headingSpoiler6">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Evenness Output
  </h3>
</button>
<div id="collapseSpoiler6" class="accordion-collapse collapse" aria-labelledby="headingSpoiler6" data-bs-parent="#accordionSpoiler6">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Foi8wbq9x4uztgkjv422gd%2Fevenness-group-significance.qzv%3Frlkey%3Dlvmydv7kkuotbtz2cilztyss7%26dl%3D1" class="external-link">Click
to view the evenness-group-significance.qzv file in QIIME 2
View.</a></p>
<p>Select “Animal” under the “Column” dropdown menu.</p>
</div>
</div>
</div>
</div>
<p>Finally, we’ll analyse sample composition in the context of
categorical metadata using a permutational multivariate analysis of
variance (PERMANOVA, first described in Anderson (2001)) test using the
<code>beta-group-significance</code> command.</p>
<p>The following commands will test whether distances between samples
within a group are more similar to each other then they are to samples
from the other groups. If you call this command with the
<code>--p-pairwise parameter</code>, as we’ll do here, it will also
perform pairwise tests that will allow you to determine which specific
pairs of groups differ from one another, if any.</p>
<p>This command can be slow to run, especially when passing
<code>--p-pairwise</code>, since it is based on permutation tests. So,
unlike the previous commands, we’ll run beta-group-significance on
specific columns of metadata that we’re interested in exploring, rather
than all metadata columns to which it is applicable.</p>
<p>Here we’ll apply this to our unweighted UniFrac distances, using two
sample metadata columns (Gender and Animal), as follows.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="at">--i-distance-matrix</span> analysis/diversity_metrics/unweighted_unifrac_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="at">--m-metadata-column</span> Sex <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> analysis/visualisations/unweighted-unifrac-gender-significance.qzv <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  <span class="at">--p-pairwise</span></span></code></pre>
</div>
<p>Copy
<code>analysis/visualisations/unweighted-unifrac-gender-significance.qzv</code>
to your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler7" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler7" aria-expanded="false" aria-controls="collapseSpoiler7">
  <h3 class="accordion-header" id="headingSpoiler7">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Gender Significance Output
  </h3>
</button>
<div id="collapseSpoiler7" class="accordion-collapse collapse" aria-labelledby="headingSpoiler7" data-bs-parent="#accordionSpoiler7">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fo8jzhit0mgs2w13i5mzno%2Funweighted-unifrac-gender-significance.qzv%3Frlkey%3Di4pddpyo9cosz7tg2dk8nr2zr%26dl%3D1" class="external-link">Click
to view the unweighted-unifrac-gender-significance.qzv file in QIIME 2
View.</a></p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="at">--i-distance-matrix</span> analysis/diversity_metrics/unweighted_unifrac_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> echidna_metadata.tsv <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">--m-metadata-column</span> Animal <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> analysis/visualisations/unweighted-unifrac-animal-significance.qzv <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>  <span class="at">--p-pairwise</span></span></code></pre>
</div>
<p>Copy
<code>analysis/visualisations/unweighted-unifrac-animal-significance.qzv</code>
to your local computer and view in QIIME 2 View (q2view).</p>
<div id="accordionSpoiler8" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler8" aria-expanded="false" aria-controls="collapseSpoiler8">
  <h3 class="accordion-header" id="headingSpoiler8">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Check Your Work: Animal Significance Output
  </h3>
</button>
<div id="collapseSpoiler8" class="accordion-collapse collapse" aria-labelledby="headingSpoiler8" data-bs-parent="#accordionSpoiler8">
<div class="accordion-body">
<p><a href="https://view.qiime2.org/visualization/?type=html&amp;src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fir3xgvn0vdwajx5hw663o%2Funweighted-unifrac-animal-significance.qzv%3Frlkey%3Dcekbzric4xww39b7hlf4x4772%26dl%3D1" class="external-link">Click
to view the unweighted-unifrac-animal-significance.qzv file in QIIME 2
View.</a></p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="provenance-and-reproducibility">Provenance and Reproducibility<a class="anchor" aria-label="anchor" href="#provenance-and-reproducibility"></a></h2>
<hr class="half-width"><p>QIIME 2 stores <strong>provenance metadata</strong> in every
<code>.qzv</code> file it generates. This allows you to track exactly
what commands, parameters, and inputs were used to create a
result—making your analysis transparent and reproducible.</p>
<p>To explore this, open any <code>.qzv</code> file (e.g.,
unweighted-unifrac-animal-significance.qzv) in QIIME 2 View and select
the Provenance tab.</p>
<p>Take note of:</p>
<ul><li>Which commands were run</li>
<li>The inputs used</li>
<li>Any parameters set</li>
<li>Intermediate files created along the way</li>
</ul><p>This is particularly useful when troubleshooting or writing up your
methods.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>##Note Click to view the <a href="https://view.qiime2.org/provenance/?src=https%3A%2F%2Fdl.dropboxusercontent.com%2Fscl%2Ffi%2Fir3xgvn0vdwajx5hw663o%2Funweighted-unifrac-animal-significance.qzv%3Frlkey%3Dcekbzric4xww39b7hlf4x4772%26dl%3D1" class="external-link">unweighted-unifrac-animal-significance.qzv
provenance file in QIIME 2 View.</a></p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>QIIME2 visualisations allow exploration of microbial composition,
diversity, and statistical differences across groups.</li>
<li>Taxonomic barplots, rarefaction curves, and diversity metrics
provide insights into microbial community structure and sampling
depth.</li>
<li>Alpha diversity measures within-sample richness and evenness, while
beta diversity measures between-sample dissimilarity.</li>
<li>Group significance tests help assess whether diversity varies across
categories like sex or animal ID.</li>
<li>Core metrics pipelines simplify the process of generating multiple
diversity outputs from a single command.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/section-3-build-a-phylogenetic-tree.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/section-5-exporting-data-for-further-analysis-in-r.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/section-3-build-a-phylogenetic-tree.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Section 3: Build a
        </a>
        <a class="chapter-link float-end" href="../instructor/section-5-exporting-data-for-further-analysis-in-r.html" rel="next">
          Next: Section 5 -...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/edit/main/episodes/section-4-basic-visualisations-and-statistics.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/rpapaluca/intro-QIIME2-mammal/" class="external-link">Source</a></p>
				<p><a href="https://github.com/rpapaluca/intro-QIIME2-mammal/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:fixme@fixme.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.12" class="external-link">sandpaper (0.16.12)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/melbournebioinformatics/uom-varnish/tree/main" class="external-link">varnish (1.0.7.9000)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "melbournebioinformatics.github.io/uom-varnish/instructor/section-4-basic-visualisations-and-statistics.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "metabarcoding, QIIME2, microbiome ",
  "name": "Section 4: Basic visualisations and statistics",
  "creativeWorkStatus": "active",
  "url": "melbournebioinformatics.github.io/uom-varnish/instructor/section-4-basic-visualisations-and-statistics.html",
  "identifier": "melbournebioinformatics.github.io/uom-varnish/instructor/section-4-basic-visualisations-and-statistics.html",
  "dateCreated": "2023-11-01",
  "dateModified": "2025-08-05",
  "datePublished": "2025-08-05"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

